x/application/keeper/transfer_applications.go

```diff
@@ -0,0 +1,241 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	cosmostypes "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/pokt-network/poktroll/telemetry"
+	"github.com/pokt-network/poktroll/x/application/types"
+	"github.com/pokt-network/poktroll/x/shared"
+)
+
+// EndBlockerTransferApplication completes pending application transfers.
+// This always happens on the last block of a session during which the transfer started.
+// It is accomplished by:
+//  1. Copying the current state of the source app onto a new destination app
+//  2. Unstaking (removing) the source app
+//  3. Staking (storing) the destination app
+func (k Keeper) EndBlockerTransferApplication(ctx context.Context) error {
+	isSuccessful := false
+	defer telemetry.EventSuccessCounter(
+		"transfer_application_end",
+		telemetry.DefaultCounterFn,
+		func() bool { return isSuccessful },
+	)
+
+	sdkCtx := cosmostypes.UnwrapSDKContext(ctx)
+	currentHeight := sdkCtx.BlockHeight()
+	sharedParams := k.sharedKeeper.GetParams(ctx)
+	logger := k.Logger().With("method", "EndBlockerTransferApplication")
+
+	// Only process application transfers at the end of the session in
+	// order to avoid inconsistent/unpredictable mid-session behavior.
+	if !shared.IsSessionEndHeight(&sharedParams, currentHeight) {
+		return nil
+	}
+
+	// Iterate over all applications and transfer the ones that have finished the transfer period.
+	// TODO_MAINNET: Use an index to iterate over the applications that have initiated
```

Comment:
Can you add your name here?

Would love it if you can start looking into indexing on-chain!

Reply:
https://github.com/pokt-network/poktroll/issues/854